src/validators/AllowedSessionsValidator.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import { Transaction } from "@matterlabs/zksync-contracts/l2/system-contracts/libraries/TransactionHelper.sol";
+
+import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { ECDSA } from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import { AccessControl } from "@openzeppelin/contracts/access/AccessControl.sol";
+import { IAccessControl } from "@openzeppelin/contracts/access/IAccessControl.sol";
+
+import { IModuleValidator } from "../interfaces/IModuleValidator.sol";
+import { IModule } from "../interfaces/IModule.sol";
+import { IValidatorManager } from "../interfaces/IValidatorManager.sol";
+import { SessionLib } from "../libraries/SessionLib.sol";
+import { Errors } from "../libraries/Errors.sol";
+import { SignatureDecoder } from "../libraries/SignatureDecoder.sol";
+
+import { SessionKeyValidator } from "./SessionKeyValidator.sol";
+
+/// @title AllowedSessionsValidator
+/// @author Oleg Bedrin - <o.bedrin@xsolla.com> - Xsolla Web3
+/// @custom:security-contact security@matterlabs.dev and o.bedrin@xsolla.com
+/// @notice This contract is used to manage allowed sessions for a smart account.
+contract AllowedSessionsValidator is SessionKeyValidator, AccessControl {
+  using SessionLib for SessionLib.SessionStorage;
+
+  /// @notice Role identifier for session registry managers.
+  bytes32 public constant SESSION_REGISTRY_MANAGER_ROLE = keccak256("SESSION_REGISTRY_MANAGER_ROLE");
+
+  /// @notice Mapping to track whether a session actions is allowed.
+  /// @dev The key is the hash of session actions, and the value indicates if the actions are allowed.
+  mapping(bytes32 sessionActionsHash => bool active) public isSessionSpecAllowed;
+
+  /// @notice Emitted when session actions are allowed or disallowed.
+  /// @param sessionActionsHash The hash of the session actions.
+  /// @param allowed Boolean indicating if the session actions are allowed.
+  event SessionActionsAllowed(bytes32 sessionActionsHash, bool allowed);
+
+  /// @notice Error indicating that the session actions are not allowed.
+  /// @param sessionActionsHash The hash of the session actions that are not allowed.
+  error SessionActionsNotAllowed(bytes32 sessionActionsHash);
```

Comment:
Please move the error into `libraries/Errors.sol`

Reply:
Gotcha.